src/main.rs

```diff
@@ -206,151 +185,20 @@ fn main() {
 
     cursive.add_fullscreen_layer(layout.with_id("main"));
 
-    // Register commands
-    cmd_manager.register(
-        "quit",
-        vec!["q", "x"],
-        Box::new(move |s, _args| {
-            s.quit();
-            Ok(None)
-        }),
-    );
-
-    {
-        let queue = queue.clone();
-        cmd_manager.register(
-            "toggleplayback",
-            vec!["toggleplay", "toggle", "play", "pause"],
-            Box::new(move |_s, _args| {
-                queue.lock().expect("could not lock queue").toggleplayback();
-                Ok(None)
-            }),
-        );
-    }
-
-    {
-        let queue = queue.clone();
-        cmd_manager.register(
-            "stop",
-            Vec::new(),
-            Box::new(move |_s, _args| {
-                queue.lock().expect("could not lock queue").stop();
-                Ok(None)
-            }),
-        );
-    }
-
-    {
-        let queue = queue.clone();
-        cmd_manager.register(
-            "previous",
-            Vec::new(),
-            Box::new(move |_s, _args| {
-                queue.lock().expect("could not lock queue").previous();
-                Ok(None)
-            }),
-        );
-    }
-
-    {
-        let queue = queue.clone();
-        cmd_manager.register(
-            "next",
-            Vec::new(),
-            Box::new(move |_s, _args| {
-                queue.lock().expect("could not lock queue").next();
-                Ok(None)
-            }),
-        );
-    }
-
-    {
-        let queue = queue.clone();
-        cmd_manager.register(
-            "clear",
-            Vec::new(),
-            Box::new(move |_s, _args| {
-                queue.lock().expect("could not lock queue").clear();
-                Ok(None)
-            }),
-        );
-    }
-
-    {
-        cmd_manager.register(
-            "queue",
-            Vec::new(),
-            Box::new(move |s, _args| {
-                s.call_on_id("main", |v: &mut ui::layout::Layout| {
-                    v.set_view("queue");
-                });
-                Ok(None)
-            }),
-        );
-    }
+    let mut cmd_manager = CommandManager::new();
+    cmd_manager.register_all(spotify.clone(), queue.clone());
 
+    #[cfg(feature = "mpris")]
     {
-        cursive.add_global_callback(Key::F1, move |s| {
-            s.call_on_id("main", |v: &mut ui::layout::Layout| {
-                v.set_view("queue");
-            });
-        });
+        let mpris_manager = mpris_manager.clone();
+        cmd_manager.register_callback(Box::new(move || {
+            mpris_manager.update();
```

Comment:
Why is an MPRIS update necessary necessary on command execution? Shouldn't the player events cover the data updates?

Reply:
Looks like it isn't. I implemented this because it was previously run on received commands, but it looks like it can be completely removed. Woops.

Reply:
Mhh.. yea, I guess we can remove this. MPRIS seems to be outdated either way.

Reply:
Disregard the last sentence, I just updated to GNOME 3.32 and that may be the reason MPRIS doesn't work anymore. Seems fine now + I removed the double update.